util/predicates/cluster_predicates.go

```diff
@@ -188,3 +189,41 @@ func ClusterUnpausedAndInfrastructureReady(logger logr.Logger) predicate.Funcs {
 	// Use any to ensure we process either create or update events we care about
 	return Any(log, createPredicates, updatePredicates)
 }
+
+// ClusterHasTopology returns a Predicate that returns true when cluster.Spec.Topology
+// is NOT nil and false otherwise.
+func ClusterHasTopology(logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return processIfTopologyManaged(logger.WithValues("predicate", "ClusterHasTopology", "eventType", "update"), e.ObjectNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return processIfTopologyManaged(logger.WithValues("predicate", "ClusterHasTopology", "eventType", "create"), e.Object)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return processIfTopologyManaged(logger.WithValues("predicate", "ClusterHasTopology", "eventType", "delete"), e.Object)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return processIfTopologyManaged(logger.WithValues("predicate", "ClusterHasTopology", "eventType", "generic"), e.Object)
+		},
+	}
+}
+
+func processIfTopologyManaged(logger logr.Logger, object client.Object) bool {
+	log := logger.WithValues("predicate", "ClusterIsManagedTopology", "eventType", "update")
+	cluster, ok := object.(*clusterv1.Cluster)
+	if !ok {
+		log.V(4).Info("Expected Cluster", "type", object.GetObjectKind().GroupVersionKind().String())
+		return false
+	}
+
+	log.WithValues("namespace", cluster.Namespace, "cluster", cluster.Name)
+
+	if cluster.Spec.Topology != nil {
+		log.V(6).Info("Cluster has topology, allowing further processing")
+		return true
+	}
+
+	log.V(4).Info("Cluster does not have topology, blocking further processing")
```

Comment:
```suggestion
	log.V(6).Info("Cluster does not have topology, blocking further processing")
```

Can you please change it to log level 6? 

In some of the other predicates we log with log level 4 because those predicates only block temporarily. This log level will be logged with every reconcile of a Cluster resource which is not using ClusterClass and it would be too verbose on log level 4.